src/parser.ts

```diff
@@ -8620,14 +8620,29 @@ export function parsePropertyDefinition(
 
   matchOrInsertSemicolon(parser, context);
 
-  return finishNode(parser, context, start, line, column, {
-    type: 'PropertyDefinition',
-    key,
-    value,
-    static: (state & PropertyKind.Static) > 0,
-    computed: (state & PropertyKind.Computed) > 0,
-    decorators
-  } as any);
+  return finishNode(
+    parser,
+    context,
+    start,
+    line,
+    column,
+    (context & Context.OptionsNext
+      ? {
+          type: 'PropertyDefinition',
+          key,
+          value,
+          static: (state & PropertyKind.Static) > 0,
+          computed: (state & PropertyKind.Computed) > 0,
+          decorators
+        }
+      : {
+          type: 'PropertyDefinition',
+          key,
+          value,
+          static: (state & PropertyKind.Static) > 0,
+          computed: (state & PropertyKind.Computed) > 0
+        }) as any
+  );
```

Comment:
These two branches are quite big, can we use 

```js
decorator: (context & Context.OptionsNext ? decorator : undefined)
```

?

Or

```js
...(context & Context.OptionsNext ? {decorator} : undefined)
``

if tge key existence matters.

Reply:
There are few cases like this. I will clean them up.